packages/webpack/mixins/fast-build/mixin.core.js

```diff
@@ -0,0 +1,45 @@
+const { Mixin } = require('hops-bootstrap');
+
+function include(rules, path) {
+  const shouldInclude =
+    // allow everything outside of node_modules
+    !path.includes('node_modules') ||
+    // allow all hops packages
+    /node_modules[/\\]hops/.test(path) ||
+    // check if any other rule matches
+    rules.some((rule) => rule.test(path));
+
+  return shouldInclude;
+}
+
+class HopsFastBuildMixin extends Mixin {
+  configureBuild(webpackConfig, { jsLoaderConfig }, target) {
+    const enabled =
+      this.options.fastBuild || (this.options.fastDev && target === 'develop');
+
+    if (!enabled) {
+      return;
+    }
+
+    process.env.__HOPS_FAST_BUILD__ = 'true';
+
+    const {
+      experimental: { fastBuild: { includeRules = [] } = {} } = {},
+    } = this.config;
+    const rules = includeRules.map((rule) => new RegExp(rule));
+    const options = jsLoaderConfig.options.presets[0][1];
```

Comment:
I'd use `[].find(…)` here to be on the save side. You'll still be able the mutate the preset's options object.

Reply:
Ok, I have changed that already but lets wait for more feedback first.

```js
    const [, options = {}] =
      jsLoaderConfig.options.presets.find(([path]) =>
        /@babel[/\\]preset-env/.test(path)
      ) || [];
```